src/app/signup/page.tsx

```diff
@@ -1,8 +1,41 @@
+'use client'
+
 import Image from "next/image";
 import { GradButton } from "@/components/ui/grad-button";
 import SignUpImage from "@/components/ui/signup-image";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
 
 const SignUp = () => {
+
+  // Example values for A and B
+  const A = 3; // Minimum length
+  const B = 30; // Maximum length
+
+  const signUpSchema = z.object({
+    fullName: z.string()
+      .min(A, { message: `Name must be at least ${A} characters long` })
+      .max(B, { message: `Name must be no more than ${B} characters long` })
+      .regex(/^[a-zA-Z'-]+$/, { message: "Name can only include alphabetical characters, hyphens, and apostrophes" }),
```

Comment:
What about spaces? If you try to type `Colton Ehrman` will it work?

Reply:
Fixed it so it allows internal spaces (no spaces at the start or at the end) as well